lib/bank-player.js

```diff
@@ -38,10 +43,10 @@ function playBuffer (ctx, destination, buffer, time, duration, gain) {
   var vca = ctx.createGain()
   source.connect(vca)
   vca.connect(destination)
-  vca.gain.linearRampToValueAtTime(gain, time)
+  vca.gain.linearRampToValueAtTime(gain, ctx.currentTime)
   if (duration > 0) {
     source.start(time, 0, duration)
-    vca.gain.exponentialRampToValueAtTime(0.1, time + duration)
+    vca.gain.exponentialRampToValueAtTime(0.01, gain, ctx.currentTime + time + duration)
```

Comment:
Ow the api is not an offset but a time?
On 20 May 2016 4:03 p.m., "danigb" <notifications@github.com> wrote:

> In lib/bank-player.js
> <https://github.com/danigb/soundfont-player/pull/25#discussion_r64045880>:
>
> >    if (duration > 0) {
> >      source.start(time, 0, duration)
> > -    vca.gain.exponentialRampToValueAtTime(0.1, time + duration)
> > +    vca.gain.exponentialRampToValueAtTime(0.01, gain, ctx.currentTime + time + duration)
>
> This is not correct. time variable should be absolute Web Audio API time.
> This is how the WAA works: if time variable is less than ctx.currentTime
> then the Web Audio API converts it to ctx.currentTime else, it uses the
> value. So, instead of ctx.currentTime + time it should be time
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/danigb/soundfont-player/pull/25/files/b04dc218444c7802da7c11da52ddd449aede71e3#r64045880>
>